server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -195,6 +222,11 @@ protected void customizeProxyRequest(Request proxyRequest, HttpServletRequest re
   protected URI rewriteURI(HttpServletRequest request)
   {
     final String host = (String) request.getAttribute(HOST_ATTRIBUTE);
+    return rewriteURI(request, host);
+  }
+
+  protected URI rewriteURI(HttpServletRequest request, String host)
```

Comment:
how would you feel about making this an actual URI builder instead of a string builder?